### PR TITLE
replaced TransitionEnd callbacks with setTimeout

### PR DESCRIPTION
**Pull request by lPadier**
In React 0.14-rc1, CSSTransitionGroup changed the way they listened to the end of an animation from listening to the EndEvent, to setting a timeout prop as listening to the end event was not reliable enough.
https://facebook.github.io/react/blog/2015/09/10/react-v0.14-rc1.html
https://github.com/facebook/react/issues/1326

Moreover, on iOS, swiping on the slick during a scroll could sometimes block the component in an animating state.

### CODE REVIEW

**Comment by vramana** [comment]
@lPadier Thanks for your PR!

**Comment by vramana** [comment]
@akiran Can you merge this PR??

**Comment by vramana** [comment]
@lPadier Actually, now that I think about it. Your PR is a bit incomplete. You should do clearTimeout of the setTimeout you are creating in componentWillUnmount. Otherwise, if you unmount mid animation it might throws errors. Please change this accordingly.

**Comment by lPadier** [comment]
@vramana good catch, thanks.
I also created a commit that has this pull request and the folders created by `npm run pre-publish` committed:
https://github.com/lPadier/react-slick/commit/e00b8991995eb8327ed7023b487ac49eca7d3843
You can use it by updating the react-slick line in your package.json to 
```json
"react-slick": "lPadier/react-slick#e00b899",
```

**Comment by vramana** [comment]
@lPadier Nice. Now this is good to go.

File: src/mixins/helpers.js [hunk 1]
```diff
@@ -206,12 +205,12 @@ var helpers = {
...
       };
 
-      callback = () => {
+      this.animationEndCallback = () => {
         this.setState(nextStateChanges);
         if (this.props.afterChange) {
           this.props.afterChange(currentSlide);
         }
-        ReactTransitionEvents.removeEndEventListener(this.refs.track.getDOMNode(), callback);
+        delete this.animationEndCallback;
```
Comment (akiran) [inline src/mixins/helpers.js]: @lPadier I think you have to use clearTimeout  and not delete callback

Comment (iam4x) [inline src/mixins/helpers.js]: He didn't named `this.animationEndCallback = window.setTimeout` so there's no need of `clearTimeout` in this case.

File: src/inner-slider.jsx [hunk 2]
```diff
@@ -50,6 +50,9 @@ export var InnerSlider = React.createClass({
     if (this.resizeListener) {
       this.resizeListener.remove();
     }
+    if (this.animationEndCallback) {
+      clearTimeout(this.animationEndCallback);
```
Comment (iam4x) [inline src/inner-slider.jsx]: This won't work.

File: src/mixins/helpers.js [hunk 3]
```diff
@@ -108,14 +107,14 @@ var helpers = {
         if (this.props.afterChange) {
           this.props.afterChange(currentSlide);
         }
-        ReactTransitionEvents.removeEndEventListener(this.refs.track.getDOMNode().children[currentSlide], callback);
+        delete this.animationEndCallback;
```
Comment (lPadier) [inline src/mixins/helpers.js]: I delete the reference to the animation timer as it is no longer necessary.
There is no need to to a clearTimeout as the timeout has already ended.

**Comment by iam4x** [comment]
Looks good to me! 

@akiran Any ETA or want some help with new contributors?

**Comment by akiran** [comment]
I will all the review all the open PR's and merge them by this weekend.

**Comment by akiran** [comment]
Incorporated this fix in latest release

**Pull request closed without merging**